test/test_build_python.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2019 Rover Robotics
+# Licensed under the Apache License, Version 2.0
+import asyncio
+from pathlib import Path
+from types import SimpleNamespace
+
+import colcon_core.shell
+from colcon_core.package_descriptor import PackageDescriptor
+from colcon_core.plugin_system import SkipExtensionException
+from colcon_core.shell.bat import BatShell
+from colcon_core.shell.sh import ShShell
+from colcon_core.subprocess import new_event_loop
+from colcon_core.task import TaskContext
+from colcon_core.task.python.build import PythonBuildTask
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def monkey_patch_get_shell_extensions(monkeypatch):
+    a_shell = None
+    for sh in [ShShell, BatShell]:
+        try:
+            a_shell = sh()
+        except SkipExtensionException:
+            pass
+
+    if a_shell is None:
+        pytest.skip()
+
+    monkeypatch.setattr(
+        colcon_core.shell,
+        'get_shell_extensions',
+        lambda: {
+            200: {'mock': a_shell}
+        }
+    )
+
+
+@pytest.fixture(autouse=True)
+def monkey_patch_put_event_into_queue(monkeypatch):
+    monkeypatch.setattr(
+        TaskContext, 'put_event_into_queue', lambda *args: None
+    )
+
+
+@pytest.fixture
+def python_build_task(tmp_path):
+    task = PythonBuildTask()
+    package = PackageDescriptor(tmp_path / 'src')
+    package.name = 'test_package'
+    package.type = 'python'
+
+    context = TaskContext(
+        pkg=package,
+        args=SimpleNamespace(
+            path=str(tmp_path / 'src'),
+            symlink_install=False,
+            build_base=str(tmp_path / 'build'),
+            install_base=str(tmp_path / 'install'),
+            merge_install=False,
+            test_result_base=str(tmp_path / 'test'),
```

Comment:
Neither of these two options `merge_install` and `test_result_base` should be necessary.